game-core/src/main/java/games/strategy/engine/data/CaseInsensitiveDynamicallyModifiable.java

```diff
@@ -0,0 +1,15 @@
+package games.strategy.engine.data;
+
+import java.util.Optional;
+
+import org.apache.commons.collections4.map.CaseInsensitiveMap;
+
+/**
+ * A realization of {@link DynamicallyModifiable} that uses case-insensitive property names.
+ */
+public interface CaseInsensitiveDynamicallyModifiable extends DynamicallyModifiable {
+  @Override
+  default Optional<MutableProperty<?>> getProperty(final String name) {
+    return Optional.ofNullable(new CaseInsensitiveMap<>(getPropertyMap()).get(name));
```

Comment:
I actually think using a new library for this is overkill. 
Simply converting the first character of the name to lowercase and the querying the map should be sufficient, because that's the only case-insensitive part that was there before.

Reply:
I wasn't sure if title case was the only variant we needed to account for because I haven't looked at all unofficial maps (I've only seen a couple of Roger Cooper's maps that indeed only use title case).  That's why I simply restored the legacy behavior.  But, if that's the only variant we need to support, then I totally agree with your recommendation.

@ron-murhammer Are you aware of maps which may use attachment option names that only differ from the standard names by the case of the first character?

Reply:
@ssoloff The legacy behaviour wasn't completely case insensitive either.
The legacy behaviour automatically converted the first character to upper case to match the getter and setter names.
Just try this using an older version

Reply:
:+1: Ah yes, I see that now in the old `PropertyUtil` code.  Not sure why I missed it before.

Reply:
It might be a good feature actually do to full case insensitive instead of just the first letter. The first letter seems like a work-around.

Re: new library - `PropertyUtil` has a few smell's. First `Util` in name, means it does not have a focus. I'm not sure if this specific functionality should be split out from `PropertyUtil`, but I suspect `PropertyUtil` would benefit from some breakup, some additional design for it. But also, such larger refactorings is probably best left out of scope for this specific update too.

Reply:
> It might be a good feature actually do to full case insensitive instead of just the first letter. The first letter seems like a work-around.

Point taken.  But remember that we're simply trying to preserve legacy behavior for the time being.  Most likely, in the next incompatible release, we would drop any kind of case-insensitive support.

@RoiEXLab @ron-murhammer Any thoughts about the two proposals?  That is, full case-insensitive support (would be new behavior) vs. title-case support (legacy behavior).

> new library - `PropertyUtil` has a few smell's

`PropertyUtil` is legacy code that was removed in #3107.  It contained the implementation previously used to do case-insensitive lookups.  It was mentioned here simply as a reference.

Reply:
> Most likely, in the next incompatible release, we would drop any kind of case-insensitive support.

Why? Seems like busy work, and opens up errors for map makers on casing that are a bit needless.

> PropertyUtil is legacy code that was removed in #3107.

I was not aware of this latest. Okay.

Reply:
This particular instance of case-insensitivity was a side effect of our use of reflection with attachments within the game parser; I don't believe it was intentional.  And I suspect it was not consistently applied across all attribute values in the XML.  I would think map makers would prefer consistency as opposed to "you can use case insensitive values here but not there."

Reply:
> I would think map makers would prefer consistency as opposed to "you can use case insensitive values here but not there."

I'm actually not sure about that. I suspect most map makers want it to work. Getting maps to work is an exercise of pulling teeth and involves many tears, anything to make it less painful would be welcome. I think it's mostly the development team that want/care about the casing consistency. I have trouble seeing why a map maker would really care. 

The real consistency issues are in redundant attachments, those that do the opposite of what they claim, all part of why map making is very difficult and requires for more skill/knowledge than it should.

Reply:
@DanVanAtta Perhaps we can just agree on restoring the old behaviour for now, and thinking about case-insensivity when actively reworking the map parsing system?